homeassistant/components/konnected/sensor.py

```diff
@@ -73,54 +86,41 @@ def async_add_ds18b20(attrs):
 class KonnectedSensor(SensorEntity):
     """Represents a Konnected DHT Sensor."""
 
-    def __init__(self, device_id, data, sensor_type, addr=None, initial_state=None):
+    def __init__(
+        self,
+        device_id,
+        data,
+        description: SensorEntityDescription,
+        addr=None,
+        initial_state=None,
+    ):
         """Initialize the entity for a single sensor_type."""
+        self.entity_description = description
         self._addr = addr
         self._data = data
-        self._device_id = device_id
-        self._type = sensor_type
         self._zone_num = self._data.get(CONF_ZONE)
-        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
-        self._unique_id = addr or f"{device_id}-{self._zone_num}-{sensor_type}"
+        self._attr_unique_id = addr or f"{device_id}-{self._zone_num}-{description.key}"
 
         # set initial state if known at initialization
         self._state = initial_state
         if self._state:
             self._state = round(float(self._state), 1)
 
         # set entity name if given
-        self._name = self._data.get(CONF_NAME)
-        if self._name:
-            self._name += f" {SENSOR_TYPES[sensor_type][0]}"
+        if name := self._data.get(CONF_NAME):
+            name += f" {description.name}"
+        self._attr_name = name
```

Comment:
Since `_attr_name` will now be defined, it will not fall back to using `description.name`

Reply:
This was the original logic. The `name` property just returned ` self._name` which could have been `None`. I would guess though that `self._data.get(CONF_NAME)` will almost always return a value.

https://github.com/home-assistant/core/blob/32b8be5a6eda4b6a14a3f97dc7c5d3305ce9909d/homeassistant/components/konnected/panel.py#L232-L239

Reply:
This is some strange logic, but I guess it "works" :man_shrugging: